recipes/gobject-introspection/all/conanfile.py

```diff
@@ -1,129 +1,180 @@
+from conan.tools.layout import basic_layout
+from conan.tools.meson import MesonToolchain, Meson
+from conan.tools.env import Environment
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.scm import Version
+from conan import conan_version
+
+required_conan_version = ">=1.60.0 <2.0 || >=2.0.5"
+
 
 class GobjectIntrospectionConan(ConanFile):
     name = "gobject-introspection"
-    description = "GObject introspection is a middleware layer between C libraries (using GObject) and language bindings"
-    topics = ("conan", "gobject-instrospection")
+    description = ("GObject introspection is a middleware layer between "
+                   "C libraries (using GObject) and language bindings")
+    license = "LGPL-2.1-or-later"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://gitlab.gnome.org/GNOME/gobject-introspection"
-    license = "LGPL-2.1"
-    settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
-    _source_subfolder = "source_subfolder"
-    _build_subfolder = "build_subfolder"
-    generators = "pkg_config"
+    topics = ("gobject-instrospection",)
 
-    @property
-    def _is_msvc(self):
-        return self.settings.compiler == "Visual Studio"
-
-    def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "build_introspection_data": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "build_introspection_data": True,
+    }
+    short_paths = True
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("%s recipe does not support windows. Contributions are welcome!" % self.name)
+        if self.settings.os in ["Windows", "Macos"] or cross_building(self):
+            # FIXME: tools/g-ir-scanner fails to load glib
+            self.options.build_introspection_data = False
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        # FIXME: g-ir-scanner fails to load glib correctly, resulting in failure during the build
+        self.options["glib"].shared = True
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/1.76.1/meson.build?ref_type=tags#L127-131
+        self.requires("glib/2.78.3", transitive_headers=True, transitive_libs=True)
+        # ffi.h exposed by plublic header gobject-introspection-1.0/girffi.h
```

Comment:
```suggestion
        # ffi.h is exposed by public header gobject-introspection-1.0/girffi.h
```

Reply:
it does not worth an entire new build for typo. In case having something substantial, I'll add all together. Thank you.